### PR TITLE
fixed ampds converter

### PR DESCRIPTION
**Pull request by klemenjak**
Hi,

I experienced some issues with the dataset AMPds2 and the new API. I found that the appliance types in the metadata were not correct, so I changed that. Also, I edited the converter for AMPds to fit the second version, which has more data.

Best,
Christoph

### CODE REVIEW

**Comment by levaphenyl** [comment]
Relates to https://github.com/nilmtk/nilmtk/issues/682

File: nilmtk/dataset_converters/ampds/convert_ampds.py [hunk 1]
```diff
@@ -46,18 +46,22 @@ def convert_ampds(input_path, output_filename, format='HDF'):
...
     --------------
-    convert('/AMPds/electricity', 'store.h5')    
+    convert_ampds('dataverse_files/', 'AMPds.h5')
 
     """
     check_directory_exists(input_path)
     files = [f for f in listdir(input_path) if isfile(join(input_path, f)) and
-             '.csv' in f and '.swp' not in f]
+             '.csv' in f and '.swp' not in f and 'Electricity' in f and 'Billing' not in f and 'Monthly' not in f
+             and  'Electricity_I.csv' not in f   and  'Electricity_Q.csv' not in f   and  'Electricity_P.csv' not in f
```
Comment (levaphenyl) [inline nilmtk/dataset_converters/ampds/convert_ampds.py:55]: `Electricity_P.csv` contains the active power for all channels already. Why ignoring it?

Comment (klemenjak) [reply nilmtk/dataset_converters/ampds/convert_ampds.py:55]: That had been part of the converter already. I cannot explain why they decided to ignore that file. What I can tell for sure is that the converter works.

Comment (levaphenyl) [reply nilmtk/dataset_converters/ampds/convert_ampds.py:55]: I wanted to check again today but just noticed: there are no more CSV files in https://dataverse.harvard.edu/dataset.xhtml?persistentId=doi:10.7910/DVN/FIE0S4

**Comment by levaphenyl** [comment]
CSV files are not anymore available for download on the [AMPds2 website](https://dataverse.harvard.edu/dataset.xhtml?persistentId=doi:10.7910/DVN/FIE0S4). This makes this converter obsolete already :(.

@klemenjak @PMeira Should we try to update to use the `.tab` files or do we remove the converter and recommend downloading the [`AMPds2.h5`](https://dataverse.harvard.edu/file.xhtml?persistentId=doi:10.7910/DVN/FIE0S4/F3LAOC&version=3.1) directly?

**Comment by PMeira** [comment]
@levaphenyl You have to select the original files from the download menu in the file list:

image

This download (ZIP file) includes all CSVs, the HDF5 file, and the equipment manuals.

The .h5 itself is kind of small (maybe resampled?), but if it indeed works and provides the same data, we can at least recommend it.

By the way, we should to keep the original converter (v1) since it is important for historical purposes and reproducibility. Nothing against adding the new one, of course.

**Comment by levaphenyl** [comment]
> @levaphenyl You have to select the original files from the download menu in the file list:
Useless alarm, sorry :facepalm:. Thank you for the hint @PMeira!

> By the way, we should to keep the original converter (v1) since it is important for historical purposes and reproducibility. Nothing against adding the new one, of course.
If I understand it well, the ideal converter would be able to deal with both v1 and v2 or we need 2 separate files and a lot of duplicated code :D.

File: nilmtk/losses.py [hunk 2]
```diff
@@ -16,6 +16,15 @@ def f1score(app_gt, app_pred):
...
     return f1_score(gt_temp, pred_temp)
 
+def mcc(app_gt, app_pred):
+    # https://scikit-learn.org/stable/modules/generated/sklearn.metrics.matthews_corrcoef.html?highlight=mcc
+    threshold = 10
+    gt_temp = np.array(app_gt)
+    gt_temp = np.where(gt_temp<threshold,0,1)
+    pred_temp = np.array(app_pred)
+    pred_temp = np.where(pred_temp<threshold,0,1)
+    return matthews_corrcoef(gt_temp, pred_temp)
```
Comment (PMeira) [inline nilmtk/losses.py:26]: `matthews_corrcoef` is missing from the imports in this file, right?

**Comment by PMeira** [comment]
> If I understand it well, the ideal converter would be able to deal with both v1 and v2 or we need 2 separate files and a lot of duplicated code :D.

@levaphenyl Yeah... In this case the differences are not that large (from @klemenjak's commits, column changes and some metadata), so a single converter looks feasible. 

Several of the converters have duplicated code, developing a base class/module with most of the common code would be useful later. That probably would be required to abstract some of the HDF5 aspects away too.

Given the current state of things, I think it would be fine to 
just rename the new `convert_ampds.py` to `convert_ampds_v2.py` if keeping both versions in the same file presents difficulties.

File: nilmtk/dataset_converters/ampds/convert_ampds.py [hunk 3]
```diff
@@ -22,15 +22,15 @@
...
                      'St': ('energy', 'apparent')}
 
-TIMESTAMP_COLUMN_NAME = "TS"
+TIMESTAMP_COLUMN_NAME = "unix_ts"
 TIMEZONE = "America/Vancouver"
 
 def convert_ampds(input_path, output_filename, format='HDF'):
     """
-    Convert AMPds R2013 as seen on Dataverse. Download the files
+    Convert AMPds2 R2016 as seen on Dataverse. Download the files
```
Comment (PMeira) [inline nilmtk/dataset_converters/ampds/convert_ampds.py:30]: It might be worth to add a comment here to clarify how to download those, e.g. https://github.com/nilmtk/nilmtk/pull/850#issuecomment-755602278

File: nilmtk/dataset_converters/ampds/convert_ampds.py (hunk 1 above)
Comment (klemenjak) [reply nilmtk/dataset_converters/ampds/convert_ampds.py:55]: It should work fine when you download the CSV files from Harvard instead of TAB. The reason I pushed this update was mainly due to updates in the metadata, where I corrected labels. Without those corrections to the metadata, AMPds2 cannot be used with NILMTK's API.

File: nilmtk/losses.py (hunk 2 above)
Comment (klemenjak) [reply nilmtk/losses.py:26]: You are right. I will resolve this problem. Thanks!
